src/apitypes.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2019 Google LLC
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import {Duplex} from 'stream';
+
+import {CancellablePromise} from './call';
+import {CallOptions} from './gax';
+import {GoogleError} from './googleError';
+import {Operation} from './longRunningCalls/longrunning';
+
+// gRPC functions return object with `.cancel()` method that can be used for
+// canceling the ongoing call.
+export interface GRPCCallResult {
+  cancel(): void;
+}
+
+// All GAX calls take RequestType and return ResultTuple, which can contain up
+// to three elements. The first element is always a response (post-processed for
+// special types of call such as pagination or long-running), the second
+// parameter is defined for paginated calls and stores the next page request
+// object, the third parameter stores raw (unprocessed) response object in cases
+// when it might be useful for users.
+export type RequestType = {};
```

Comment:
Any reason why we're not using generics here?

Reply:
Not sure about generics, but I think we can use `protobufjs.Type` instead of `{}` everywhere. Let me leave this for my next typings update (that will happen soon),

Reply:
To answer your specific question of why we're not using generics here, it's because I did not want to make any serious changes to types in scope of this PR which is too huge to be carefully reviewed :)